runtime/onert/core/src/loader/train/CheckpointLoader.cc

```diff
@@ -0,0 +1,108 @@
+/*
+ * Copyright (c) 2024 Samsung Electronics Co., Ltd. All Rights Reserved
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include "loader/train/CheckpointLoader.h"
+
+#include "exec/Execution.h"
+#include "ir/train/Checkpoint.h"
+#include "ir/train/TrainingInfo.h"
+
+#include <fstream>
+#include <filesystem>
+
+namespace
+{
+
+using namespace onert;
+using namespace ir;
+using namespace train;
+using namespace checkpoint;
+using namespace exec;
+
+class CheckpointLoader
+{
+public:
+  CheckpointLoader(const std::string &filename)
+  {
+    if (filename.empty() || !std::filesystem::exists(filename))
+      throw std::runtime_error{"Invalid checkpoint file"};
+
+    _file.open(filename.c_str(), std::ios::binary | std::ios::in);
+    if (!_file.good())
+      throw std::runtime_error{"Failed to open checkpoint file"};
+
+    _file.seekg(0, std::ios::end);
+    const auto filesize = _file.tellg();
+    _file.seekg(0, std::ios::beg);
+
+    if (filesize < static_cast<long long>(sizeof(_header)))
```

Comment:
For curiosity, Why is casting to a `long long` type necessary?

I checked `tellg()` return `std::streampos` type, but I couldn't catch why it should compared with `long long` type.

Reply:
@zetwhite 

AFAIK, `std::streampos` is larger than `int32_t`. So I cast `sizeof(_header)` as `long long` type. However, casting is not necessary in this case. So I removed it. :)

Reply:
> AFAIK, std::streampos is larger than int32_t.

Aha, I didn't know it. :smile: 
Since I was just curious, I don't want you to fix it. 

Anyway looks good to me.